tooling/minimize/src/rvalue.rs

```diff
@@ -20,60 +20,54 @@ impl<'cx, 'tcx> FnCtxt<'cx, 'tcx> {
                 use smir::BinOp::*;
-                let op = match *bin_op {
-                    Offset => BinOp::PtrOffset { inbounds: true },
+                let op = match (bin_op, lty) {
+                    (Offset, Type::Ptr(_)) => BinOp::PtrOffset { inbounds: true },
                     // all int ops
-                    Add => BinOp::Int(BinOpInt::Add),
-                    Sub => BinOp::Int(BinOpInt::Sub),
-                    Mul => BinOp::Int(BinOpInt::Mul),
-                    Div => BinOp::Int(BinOpInt::Div),
-                    Rem => BinOp::Int(BinOpInt::Rem),
+                    (Add, Type::Int(_)) => BinOp::Int(BinOpInt::Add),
+                    (Sub, Type::Int(_)) => BinOp::Int(BinOpInt::Sub),
+                    (Mul, Type::Int(_)) => BinOp::Int(BinOpInt::Mul),
+                    (Div, Type::Int(_)) => BinOp::Int(BinOpInt::Div),
+                    (Rem, Type::Int(_)) => BinOp::Int(BinOpInt::Rem),
+                    (Shl, Type::Int(_)) => BinOp::Int(BinOpInt::Shl),
+                    (Shr, Type::Int(_)) => BinOp::Int(BinOpInt::Shr),
+                    (BitAnd, Type::Int(_)) => BinOp::Int(BinOpInt::BitAnd),
+                    (BitOr, Type::Int(_)) => BinOp::Int(BinOpInt::BitOr),
+                    (BitXor, Type::Int(_)) => BinOp::Int(BinOpInt::BitXor),
 
-                    Lt => BinOp::IntRel(IntRel::Lt),
-                    Le => BinOp::IntRel(IntRel::Le),
-                    Gt => BinOp::IntRel(IntRel::Gt),
-                    Ge => BinOp::IntRel(IntRel::Ge),
-                    Eq => BinOp::IntRel(IntRel::Eq),
-                    Ne => BinOp::IntRel(IntRel::Ne),
+                    (Lt, Type::Int(_)) => BinOp::IntRel(IntRel::Lt),
+                    (Le, Type::Int(_)) => BinOp::IntRel(IntRel::Le),
+                    (Gt, Type::Int(_)) => BinOp::IntRel(IntRel::Gt),
+                    (Ge, Type::Int(_)) => BinOp::IntRel(IntRel::Ge),
+                    (Eq, Type::Int(_)) => BinOp::IntRel(IntRel::Eq),
+                    (Ne, Type::Int(_)) => BinOp::IntRel(IntRel::Ne),
 
-                    // implemented for int and bool
-                    BitAnd =>
-                        match self.translate_ty_smir(lty, span) {
-                            Type::Int(_) => BinOp::Int(BinOpInt::BitAnd),
-                            Type::Bool => BinOp::Bool(BinOpBool::BitAnd),
-                            ty => rs::span_bug!(span, "BitAnd not supported for: {ty:?}"),
-                        },
-                    x => rs::span_bug!(span, "Binary Op not supported: {x:?}"),
+                    // all bool ops
+                    (BitAnd, Type::Bool) => BinOp::Bool(BinOpBool::BitAnd),
+                    (BitOr, Type::Bool) => BinOp::Bool(BinOpBool::BitOr),
+                    (BitXor, Type::Bool) => BinOp::Bool(BinOpBool::BitXor),
+
+                    (op, ty) =>
+                        rs::span_bug!(span, "Binary Op {op:?} not supported for type {ty:?}"),
```

Comment:
```suggestion
                        rs::span_bug!(span, "Binary Op {op:?} not supported for type {ty}"),
```
The debug rendering of types can be quite verbose

Reply:
Sadly I cannot do this as types do not implement `std::fmt::Display`. So for now I dropped printing the types at all.

Reply:
Rust types implement `Display`... can we print the Rust type instead?